apps/src/gamelab/GameLabVisualizationColumn.jsx

```diff
@@ -37,6 +41,19 @@ var GameLabVisualizationColumn = React.createClass({
     };
   },
 
+  componentWillReceiveProps(nextProps) {
+    // Use jQuery to turn on and off the grid since it lives in a protected div
+    if (nextProps.showGrid !== this.props.showGrid) {
+      if (nextProps.showGrid) {
+        $("#grid-checkbox")[0].className = "fa fa-check-square-o";
+        $("#grid-overlay")[0].style.display = '';
+      } else {
+        $("#grid-checkbox")[0].className = "fa fa-square-o";
+        $("#grid-overlay")[0].style.display = 'none';
+      }
+    }
+  },
```

Comment:
Looking at this code again, I'm actually pretty happy with it. The pattern still feels fairly React like. One thing you could do is use refs instead of ids for grid-checkbox. Then you would have something like `this.refs.gridCheckbox.className = "fa-check-square-o".` For grid-overlay, that may not be be possible because grid-overlay is in a different component.

I think that may be what I actually like least about this - modifying the rendered DOM of another component from this one. Could/should we have a method like this in GridOverlay as well? There I think it could just be
```js
componentWillReceiveProps(nextProps) {
  $(ReactDOM.findDOMNode(this)).toggle(nextProps.showGrid)
}